### PR TITLE
feat(emp-tools): add ability to list perps and different emp versions

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David <david@umaproject.org>

# Motivation
#204

# Details
All this does is add ability to list different types of UMA contracts, as well as query for different abi verisons. This is step 1 in the process to add perpetual support.

image

### CODE REVIEW

File: features/contract-selector/ContractSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import styled from "styled-components";
```
Comment (daywiss) [inline features/contract-selector/ContractSelector.tsx:1]: This is an exact copy of EmpSelector with just a couple elements changed

Comment (nicholaspai) [reply features/contract-selector/ContractSelector.tsx:1]: What changed? I wonder if we can refactor the shared logic into a FinancialContractSelector file

File: pages/index.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import MenuIcon from "@material-ui/icons/Menu";
 import Header from "../features/core/Header";
 import ContractState from "../features/contract-state/ContractState";
 import ManagePosition from "../features/manage-position/ManagePosition";
-import EmpSelector from "../features/emp-selector/EmpSelector";
```
Comment (nicholaspai) [inline pages/index.tsx:21]: Is this file deprecated now?

**Review by nicholaspai — COMMENTED** [review]
First pass looks great really impressed with all the code you added.

File: pages/index.tsx (hunk 2 above)
Comment (daywiss) [reply pages/index.tsx:21]: Yea its deprecated. Should i just delete it?

Comment (nicholaspai) [reply pages/index.tsx:21]: Yeah why not

**Review by mrice32 — APPROVED** [review]
LGTM!

File: features/contract-selector/ContractSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  useMediaQuery,
+  InputBase,
+  MenuItem,
+  FormControl,
+  ListItemText,
+  Select,
+} from "@material-ui/core";
+import { withStyles, useTheme } from "@material-ui/core/styles";
+
+import useContractList from "./useContractList";
```
Comment (daywiss) [inline features/contract-selector/ContractSelector.tsx:13]: previously empSelector imported useEmpList, this now uses "contract" list

File: features/contract-selector/ContractSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  }
+`;
+
+const ContractSelector = () => {
+  const theme = useTheme();
+  const largeScreen = useMediaQuery(theme.breakpoints.up("sm"));
+
+  const { signer } = Connection.useContainer();
+  const { empAddress, setEmpAddress } = EmpAddress.useContainer();
+  const { contracts, loading } = useContractList();
```
Comment (daywiss) [inline features/contract-selector/ContractSelector.tsx:44]: this used to instanciate the empList now does general contracts

File: features/contract-selector/ContractSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            </MenuItem>
+          )}
+          {contracts.map((contract) => {
+            return (
+              <MenuItem value={contract.address} key={contract.address}>
+                <ListItemText
+                  primary={
+                    largeScreen
+                      ? `(${contract.type}) ${contract.name}`
+                      : `(${contract.type}) ${contract.symbol}`
```
Comment (daywiss) [inline features/contract-selector/ContractSelector.tsx:91]: this used to not include the contract type in parens, like (Perpetutal) or (EMP)

**Pull request merged**